### PR TITLE
Switch to Libtask.TapedTask

### PR DESCRIPTION
**Pull request by rikhuijzer**
Similar to https://github.com/TuringLang/Turing.jl/pull/1778.

### CODE REVIEW

**Review by devmotion — COMMENTED** [review]
Can we use `task` instead of `ttask`? Seems a bit simpler if we change the name.

**Comment by rikhuijzer** [comment]
> Can we use `task` instead of `ttask`? Seems a bit simpler if we change the name.

Libtask uses `ttask` to refer to objects of type `TapedTask`. This object also contains a field `task::Task`, so that's why separate variable names are used.

**Comment by devmotion** [comment]
Yeah but in `Trace` we don't have any other task. And the type doesn't seem very relevant from the perspective of AdvancedPS.

File: Project.toml [hunk 1]
```diff
@@ -14,7 +14,7 @@ StatsFuns = "4c63d2b9-4356-54db-8cca-17b64c39e42c"
 [compat]
 AbstractMCMC = "2, 3"
 Distributions = "0.23, 0.24, 0.25"
-Libtask = "0.6.7"
+Libtask = "0.7"
```
Comment (devmotion) [inline Project.toml:17]: The compat shouldn't be changed:

```suggestion
Libtask = "0.6.7"
```

File: Project.toml [hunk 2]
```diff
@@ -1,7 +1,7 @@
 name = "AdvancedPS"
 uuid = "576499cb-2369-40b2-a588-c64705576edc"
 authors = ["TuringLang"]
-version = "0.3.0"
+version = "0.3.4"
```
Comment (yebai) [inline Project.toml:4]:
```suggestion
version = "0.4"
```
the `master` branch contains breaking changes that have not been released yet. The most recent `v0.3x` releases are based on the [`releases` ](https://github.com/TuringLang/AdvancedPS.jl/tree/releases) branch.

**Review by yebai — APPROVED** [review]

**Comment by rikhuijzer** [comment]
@yebai could you manage the release or  provide more details on how the release process is going here at AdvancedPS?

**Comment by yebai** [comment]
> @yebai could you manage the release or provide more details on how the release process is going here at AdvancedPS?

Fixed by https://github.com/TuringLang/AdvancedPS.jl/pull/44

**Pull request merged**